administrator/components/com_content/controllers/article.php

```diff
@@ -144,6 +144,41 @@ public function batch($model = null)
 		return parent::batch($model);
 	}
 
+	public function shareTokenGenerate($length = 16)
+	{
+		jimport( 'joomla.user.helper' );
+		$salt = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+		$len = strlen($salt);
+		$token = '';
+
+		$stat = @stat(FILE);
+
+		//if(empty($stat)!==is_array($stat))
+		$stat = array(php_uname());
+		mt_srand(crc32(microtime() . implode('|', $stat)));
+		for ($i = 0; $i < $length; $i ++)
+		{
+			$token .= $salt[mt_rand(0, $len -1)];
+		}
+		return $token;
+	}
+
+	public function shareDraft()
+	{
+		// Get the model
+		$model = $this->getModel();
+		$token = $this->shareTokenGenerate();
+		$return = $model->shareToken($token);
+		if($return)
+		{
+			echo "token saved";
```

Comment:
@nikitadhiman Language strings should not be hardcoded but use JText::_('COM_CONTENT_XXX') to create language strings. Add the language strings to the appropriate language file.